src/PMGGenverter.cpp

```diff
@@ -0,0 +1,99 @@
+            m_status.inverter_temperature = base::Temperature::fromCelsius(
+                fromBigEndian<uint16_t>(msg.data + 6) / 10.0);
+            break;
+        case 0x201:
+            m_status.status = fromBigEndian<uint16_t>(msg.data);
+            m_status.inverter_alarm = fromBigEndian<uint16_t>(msg.data + 2);
+            m_status.inverter_warning = fromBigEndian<uint16_t>(msg.data + 4);
+            m_status.engine_alarm = fromBigEndian<uint16_t>(msg.data + 6);
+            break;
+        case 0x202:
+            m_status.stepper = static_cast<float>(fromBigEndian<uint16_t>(msg.data));
+            m_status.oil_temperature = base::Temperature::fromCelsius(
+                fromBigEndian<uint16_t>(msg.data + 2) / 10.0);
+            m_status.delta_dc_bus =
+                static_cast<float>(fromBigEndian<uint16_t>(msg.data + 4));
+            m_status.PWM_scale =
+                static_cast<float>(fromBigEndian<uint16_t>(msg.data + 6));
+            break;
+        case 0x203:
+            m_status.TEST_ramp = msg.data[7];
+            break;
+        case 0x204:
+            m_status.inverter_model =
+                std::string(1, msg.data[0]) + std::string(1, msg.data[1]);
+            m_status.firmware_version = static_cast<int>(msg.data[2]);
+            m_status.firmware_subversion = static_cast<int>(msg.data[3]);
+            m_status.hardware_version = static_cast<int>(msg.data[4]);
+            m_status.hardware_subversion = static_cast<int>(msg.data[5]);
+            m_status.inverter_serial_number =
+                std::string(1, msg.data[6]) + std::string(1, msg.data[7]);
+            break;
+        case 0x205:
+            uint16_t total_hours = fromBigEndian<uint16_t>(msg.data);
+            uint16_t total_minutes = fromBigEndian<uint16_t>(msg.data + 2);
+            uint16_t maintenance_hours = fromBigEndian<uint16_t>(msg.data + 4);
+            uint16_t maintenance_minutes = fromBigEndian<uint16_t>(msg.data + 6);
+            m_status.total_work =
+                base::Time::fromSeconds((total_hours * 3600) + (total_minutes * 60));
+            m_status.maintenance_work = base::Time::fromSeconds(
+                (maintenance_hours * 3600) + (maintenance_minutes * 60));
+            m_has_full_update = true;
+            break;
+    }
+}
+
+PMGGenverterStatus PMGGenverter::getStatus() const
+{
+    return m_status;
+}
+
+bool PMGGenverter::hasFullUpdate() const
+{
+    return m_has_full_update;
+}
+
+void PMGGenverter::resetFullUpdate()
+{
+    m_status = PMGGenverterStatus();
+    m_status.time = base::Time();
+    m_has_full_update = false;
+}
+
+canbus::Message PMGGenverter::queryGeneratorCommand(bool start, bool stop)
+{
+    canbus::Message msg;
+    msg.can_id = 0x210;
+    msg.size = 8;
+    msg.data[0] = start;
+    msg.data[1] = stop;
+
+    for (int i = 2; i < 7; i++) {
+        msg.data[i] = 0;
+    }
+
+    m_run_ramp = (m_run_ramp + 1) % 256;
```

Comment:
It's nice to be cute with modulos and stuff, but then m_run_ramp is a uint8_t so its maximum value is 255.

Either go back to the if/else, or change to uint16_t (ok for me either way).